packages/api/src/providers/mapper/mapper.provider.ts

```diff
@@ -0,0 +1,39 @@
+import { Mapper } from './mapper';
+
+/**
+ * A utility class that provides and manages singleton instances of `Mapper` objects.
+ *
+ * The `MapperProvider` is designed to handle the creation and reuse of `Mapper` instances
+ * for different types. Each type of `Mapper` is stored in a map, ensuring that only one instance
+ * per type exists during the application's lifecycle.
+ */
+export class MapperProvider {
+  /**
+   * A static map that stores singleton instances of `Mapper` objects, keyed by their type name.
+   */
+  // eslint-disable-next-line @typescript-eslint/consistent-generic-constructors, @typescript-eslint/no-explicit-any
+  private static readonly MAPPER_INSTANCES: Map<string, Mapper<any>> = new Map();
+
+  /**
+   * Retrieves a Mapper instance for the given type. If no instance exists, a new one is created,
+   * stored, and then returned.
+   *
+   * @template T - The specific Mapper type to retrieve or create.
+   * @param type - The constructor of the Mapper to retrieve or create.
+   * @returns The Mapper instance for the given type.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public static get<T extends Mapper<any>>(type: new () => T): T {
```

Comment:
Here it's better to be avoided usage of `any`. You can use `unknown` instead, but the best would be if you create a custom type for the mapper constructor
`type MapperConstructor<T> = new () => Mapper<T>`
and use it here
`public static get<T>(type: MapperConstructor<T>): Mapper<T> {}`

Reply:
Done